FLT/Basic/Reductions.lean

```diff
@@ -35,19 +35,136 @@ open WeierstrassCurve
 theorem Mazur_Frey (P : FreyPackage) :
     haveI : Fact P.p.Prime := ⟨P.pp⟩
     GaloisRep.IsIrreducible (P.freyCurve.galoisRep P.p P.hppos) :=
-  sorry
+  knownin1980s
 
 /-!
 
 But it follows from a profound theorem of Ribet, and the even more profound theorem
 (proved by Wiles) that the representation cannot be irreducible.
 
 -/
-
+/-- The natural `ℤ_p`-algebra structure on `ℤ/pℤ`. -/
+noncomputable local instance (p : ℕ) [Fact p.Prime] : Algebra ℤ_[p] (ZMod p) :=
+  RingHom.toAlgebra PadicInt.toZMod
+
+set_option linter.unusedVariables false in
+theorem irreducible_of_isCompatible_iff {K : Type*} [Field K] [NumberField K]
+  (E : Type*) [Field E] [NumberField E] (T : GaloisRepFamily K E 2) (comp : T.isCompatible)
+  {p : ℕ} [Fact p.Prime] {q : ℕ} [Fact q.Prime]
+  (φ : E →+* AlgebraicClosure ℚ_[p]) (ψ : E →+* AlgebraicClosure ℚ_[q]) :
+  GaloisRep.IsIrreducible (T _ φ) ↔ GaloisRep.IsIrreducible (T _ ψ) := knownin1980s
+
+open GaloisRepresentation in
 theorem Wiles_Frey (P : FreyPackage) :
-    haveI : Fact P.p.Prime := ⟨P.pp⟩
-    ¬ GaloisRep.IsIrreducible (P.freyCurve.galoisRep P.p P.hppos) :=
-  FreyCurve.torsion_not_isIrreducible P
+  haveI : Fact P.p.Prime := ⟨P.pp⟩
```

Comment:
I made commit [42e3581](https://github.com/ImperialCollegeLondon/FLT/pull/761/commits/42e358162e1d1a73cd7a2d405e1b0ddbc387d0c2) to this PR which adds comments to the proof (all long proofs should be commented -- certainly anything this long) and changes various things to be more idiomatic. Unfortunately your indentation was wrong (your statement is indented 2 and your proof 4 instead of the other way around) so it's harder to see in the diff what changes I made. Other than the comments, the main things are `rw[` -> `rw [`, `have X := blah; rcases X with \<a,b,c,...\>` -> `obtain \<a,b,c,...\> := blah`, `rw [X], rw [Y]` -> `rw [X, Y]`, and don't name instances especially if you don't refer to them.